src/adagrams.js

```diff
@@ -1,15 +1,187 @@
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
+const LETTER_SCORES = {
+  A: 1,
+  B: 3,
+  C: 3,
+  D: 2,
+  E: 1,
+  F: 4,
+  G: 2,
+  H: 4,
+  I: 1,
+  J: 8,
+  K: 5,
+  L: 1,
+  M: 3,
+  N: 1,
+  O: 1,
+  P: 3,
+  Q: 10,
+  R: 1,
+  S: 1,
+  T: 1,
+  U: 1,
+  V: 4,
+  W: 4,
+  X: 8,
+  Y: 4,
+  Z: 10,
+};
+// my copy
+
+// helper function to build weighted letter pool
+const makeWeightedPool = () => {
+  const weightedPool = [];
+  for (let letter in LETTER_POOL) {
+    let letterFreq = LETTER_POOL[letter];
+    // push letter to weightedPool, repeat letterFreq times
+    weightedPool.push(...letter.repeat(letterFreq));
+  }
+  return weightedPool;
+};
+
+////////// WAVE 1 //////////
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  // call helper function
+  let weightedPool = makeWeightedPool();
+
+  // fill randomIndices with 10 random indices from weightedPool
+  const randomIndices = new Set();
+  while (randomIndices.size < 10) {
+    let randomIndex = Math.floor(Math.random() * weightedPool.length);
+    randomIndices.add(randomIndex);
+  }
+
+  // use the random indices to get letters
+  const lettersInHand = [];
+  for (let index of randomIndices) {
+    lettersInHand.push(weightedPool[index]);
+  }
+
+  return lettersInHand;
 };
```

Comment:
Rather than creating two separate scopes, you could have created a `for loop` and combined these into one. With a `for loop` you could have used `i` as an index and avoided the additional loop. Remember we should keep the algorithmic complexity in mind.